### DISCUSSION TITLE
Issues with installing omnet 6.0.3 on a fresh Ubuntu VM

### DISCUSSION THREAD

**Discussion by anoikob** [omnetpp/omnetpp#1205] [Q&A] [answered]
Greetings.

Before I describe my issue, a disclaimer: I am a complete noob when it comes to Linux, this might be the first time I choose to do something serious with it.

I've been trying to install omnet 6.0.3 on my VM Virtualbox that contains a fresh Ubuntu 24.04.1 LTS installation. Ultimately my goal is to install INET and NeSTiNg to simulate TSN. Problem is, I am following the installation guide that came with the omnet download (yes, I've downloaded the Linux version of omnet) and I am already having issues while trying to install the prerequisite packages. To be more specific, I'm trying to do this:
![image](https://github.com/user-attachments/assets/a7f196bf-db6c-485e-918c-098be062ff5d)

But instead, in the terminal, I'm met with this:
![image](https://github.com/user-attachments/assets/ffe9c52b-e356-4114-b1e5-2dfe62279722)

Remember, this is a fresh Ubuntu installation, I have downloaded zero packages that may be considered essential, and maybe those are needed to download the omnet prerequisites. Maybe those are not needed, idk, as I said this is my first time actually doing something serious with this OS.

Later on in the manual, I'm given the option to download the packages with GUI, using something called Synaptic, which again, my Ubuntu installation does not have. This is what the manual says: 
![image](https://github.com/user-attachments/assets/1057095b-e5aa-4981-99ff-635a424cf6ff)

Yet this is what comes up when I type Synaptic in dash:
![image](https://github.com/user-attachments/assets/cb80016e-04b7-4584-8723-aaca2f5231f2)

So....what do I do? Is the manual useless? Am I missing important packages that should be there? Am I doomed?
Many thanks in advance!!!

**Comment by torokati44** [comment, marked as answer]
It seems that for Ubuntu 24.04, that package got its name changed to `libwebkit2gtk-4.1-0`. Could you try changing `libwebkit2gtk-4.0-37` in the failing command to that? The manual for the 6.0.3 release was written before Ubuntu 24.04, that's why the package name is outdated.

**Reply by anoikob** [reply]
I will try that, thanks a ton! How can I similarly find if packages have changed their names?

Edit: It worked, I installed what I needed to. Now to troubleshoot the second command (this is getting out of hand...)

**Reply by torokati44** [reply]
If there were any more, those would also have been listed in the error above. But I checked, and there was only this one.

One more thing that has changed with Ubuntu 24.04, is that the command starting with `python3 -m pip install` won't succeed out of the box. You'll probably have to add the `--break-system-packages` flag at the end of that command. I know this sounds scary, but it will Probably Be Fine.
The upcoming 6.1 release will fix these issues, and will also come with an `install.sh` script that automatizes the whole setup process.

**Reply by anoikob** [reply]
Yup, this is exactly what happened with python3 as well, I had to use that command and now omnet 6.0.3 is configured and running on my Ubuntu VM. Thanks a lot for your help!

**Reply by rhornig** [reply]
Just to add a tip for future readers: If you are installing an UbunutVM on Windows just use OMNeT++ there is a much straightforward way using the Windows Subsystem for Linux:

See: https://github.com/omnetpp/opp_env?tab=readme-ov-file#installing-opp_env-on-windows-10--11-with-wsl2

This will install an Ubuntu VM in WSL along with `opp_env` and opp_env can be used to install any version of OMNeT++ and almost all omnet models automatically without any user intervention (including installing external dependencies).
